File: src/com/marcherdiego/events/navigator/FileFilter.java
package com.marcherdiego.events.navigator;

import com.intellij.openapi.vfs.VirtualFile;
import com.intellij.psi.*;
import com.intellij.usages.ReadWriteAccessUsageInfo2UsageAdapter;
import com.intellij.usages.Usage;

public class FileFilter implements Filter {

    private final VirtualFile file;

    FileFilter(PsiMethod method) {
        this.file = method.getContainingFile().getVirtualFile();
    }

    @Override
    public boolean shouldShow(Usage usage) {
        VirtualFile usageFile = ((ReadWriteAccessUsageInfo2UsageAdapter) usage).getFile();
        return !usageFile.equals(file);
    }
}

File: src/com/marcherdiego/events/navigator/FileFilter.kt
package com.marcherdiego.events.navigator

import com.intellij.openapi.vfs.VirtualFile
import com.intellij.usages.ReadWriteAccessUsageInfo2UsageAdapter
import com.intellij.usages.Usage
import com.intellij.usages.UsageInfo2UsageAdapter

class FileFilter internal constructor(virtualFile: VirtualFile) : Filter {
    private val file: VirtualFile = virtualFile

    override fun shouldShow(usage: Usage?): Boolean {
        if (usage == null) {
            return false
        }
        return when (usage) {
            is ReadWriteAccessUsageInfo2UsageAdapter -> usage.file != file && isNotImport(usage)
            is UsageInfo2UsageAdapter -> usage.file != file && isNotImport(usage)
            else -> false
        }
    }

    private fun isNotImport(usage: Usage) = usage.toString().contains("|import|").not()
}
